src/lib/VirtualScroller.js

```diff
@@ -289,8 +289,8 @@ class VirtualScroller {
             newStartOffset = curEndOffset + 1;
             // Create elements from curEnd + 1 to newEndOffset
             this.createItems(fragment, newStartOffset, newEndOffset);
-            // Delete the elements from curStartOffset to newStartOffset
-            this.deleteItems(this.listEl, curStartOffset - curStartOffset, offset - curStartOffset);
+            // Delete the first (offset - curStartOffset) elements that scrolled out of view
+            this.deleteItems(this.listEl, 0, offset - curStartOffset);
```

Comment:
I looked at this for 20 minutes trying to figure out if there was any possible reason why 'curStartOffset - curStartOffset' was used instead of 0 and came up with nothing. Good catch.